### PR TITLE
Fast api endpoint

### PR DESCRIPTION
**Pull request by Marvin-Deng**
- Adjust the output format of extracted info into a json object to identify info with the url the data was pulled from
- Add a fast api endpoint which calls the crawler and return the extracted info
- Add time testing script to test the performance of the algorithm and adjust the output format

image

### CODE REVIEW

File: crawler/crawler.py
Comment (Michaelyya) [inline crawler/crawler.py:1]: at line 71 - "if url not in self.visited:" is redundant tho

Before BFS while loop, should've added return "self.curr_depth <= 0", it'll just avoid more actions to queue in loops

**Review by Michaelyya — CHANGES_REQUESTED** [review]
All Good! Everything looks nice and clean!

File: crawler/crawler.py
Comment (Marvin-Deng) [reply crawler/crawler.py:1]: Good catch! For the curr_depth part, I'm a bit confused on what you mean. On line 51, I have a condiiton checking for self.curr_depth > 0, which would prevent the loop from running if the depth was set to negative

image

Comment (Michaelyya) [reply crawler/crawler.py:1]: Oh yea, I see! while loop is all good tho, sorry for the confusion!! Great work, thank you so much :)))

**Review by Michaelyya — APPROVED** [review]

**Pull request merged**